core/settings/qPrefDiveComputer.h

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef QPREFSDIVECOMPUTER_H
+#define QPREFSDIVECOMPUTER_H
+#include "core/pref.h"
+
+#include <QObject>
+
+class qPrefDiveComputer : public QObject {
+	Q_OBJECT
+	Q_PROPERTY(QString device READ device WRITE set_device NOTIFY device_changed);
+	Q_PROPERTY(QString device_name READ device_name WRITE set_device_name NOTIFY device_name_changed);
+	Q_PROPERTY(int download_mode READ download_mode WRITE set_download_mode NOTIFY download_mode_changed);
+	Q_PROPERTY(QString product READ product WRITE set_product NOTIFY product_changed);
+	Q_PROPERTY(QString vendor READ vendor WRITE	set_vendor NOTIFY vendor_changed);
+
+public:
+	qPrefDiveComputer(QObject *parent = NULL);
+	static qPrefDiveComputer *instance();
+
+	// Load/Sync local settings (disk) and struct preference
+	void loadSync(bool doSync);
+	void load() {loadSync(false); }
+	void sync() {loadSync(true); }
+
+public:
+	static inline const QString device() {return QString(prefs.dive_computer.device); };
+	static inline const QString device_name() {return QString(prefs.dive_computer.device_name); };
+	static inline int download_mode() {return prefs.dive_computer.download_mode; };
+	static inline const QString product() {return QString(prefs.dive_computer.product); };
```

Comment:
1) Note that AFAIK this could be simply written as `return prefs.dive_computer.product;`. I don't see what useful information one gets from the explicit conversion constructor.

2) You return a `QString` by *value* (i.e a copy). There is no point in making it `const`. Note that although it looks similar, this is very different from `const char *product()`. The pointer you return in that case is *not* const, but the data you point to *is* const. What you have written would be the equivalent to `const char * const product()`. Note the two `const`. If you would return by reference, then const could make sense: `const QString &product()`, because there the caller does not get a unique copy, but a possibly shared reference. [Returning a reference in this case would of course be a bug - the referenced object is immediately destroyed on function return. I'm just using this as an example].

To make a long story short. If you write
```
	static inline const QString product() {...} 
```
you also have to write
```
	static inline const int download_mode() {...}
```
and, frankly, nobody would do that for good reasons.

Reply:
1. yes it could but I am no fan a of automatic conversions, I have spent too many hours debugging why “automatic” messed up my code. 
2. no need to spent time learning my basic compiler functions, I know what I did and did it for a reason (clang generates different code in the caller). changed

added new commits to make the same change in other qPref classes